src/Uno.UI/LayoutProvider.Android.cs

```diff
@@ -72,28 +83,42 @@ private void MeasureLayout(PopupWindow sender)
 			NavigationBarRect = new Rect(0, adjustNothingFrame.Bottom, realMetrics.WidthPixels, realMetrics.HeightPixels);
 
 			LayoutChanged?.Invoke(StatusBarRect, KeyboardRect, NavigationBarRect);
+		}
 
-			T Get<T>(Action<T> getter) where T : new()
-			{
-				var result = new T();
-				getter(result);
-
-				return result;
-			}
+		private void MeasureInsets(PopupWindow sender, WindowInsets insets)
+		{
+			Insets = new Thickness(
+				ViewHelper.PhysicalToLogicalPixels(insets.SystemWindowInsetLeft),
+				ViewHelper.PhysicalToLogicalPixels(insets.SystemWindowInsetTop),
+				ViewHelper.PhysicalToLogicalPixels(insets.SystemWindowInsetRight),
+				ViewHelper.PhysicalToLogicalPixels(insets.SystemWindowInsetBottom)
+			);
+
+			InsetsChanged?.Invoke(Insets);
 		}
 
+		T Get<T>(Action<T> getter) where T : new()
```

Comment:
You should decor it with `private`, if you are moving this helper method out of local scope. But, I would keep it local only, since only `MeasureLayout` uses it